android/java/org/chromium/chrome/browser/playlist/PlaylistHostActivity.java

```diff
@@ -0,0 +1,492 @@
+                            mPlaylistService.reorderItemFromPlaylist(
+                                    playlistItemModel.getPlaylistId(), playlistItemModel.getId(),
+                                    (short) i);
+                        }
+                        if (playlistItems.size() > 0) {
+                            loadPlaylist(playlistItems.get(0).getPlaylistId());
+                        }
+                    }
+                });
+
+        mPlaylistViewModel.getDeletePlaylistItems().observe(
+                PlaylistHostActivity.this, playlistItems -> {
+                    if (mPlaylistService != null) {
+                        for (PlaylistItemModel playlistItem : playlistItems.getItems()) {
+                            mPlaylistService.removeItemFromPlaylist(
+                                    playlistItems.getId(), playlistItem.getId());
+                        }
+                        loadPlaylist(playlistItems.getId());
+                    }
+                });
+
+        mPlaylistViewModel.getMoveOrCopyItems().observe(
+                PlaylistHostActivity.this, moveOrCopyModel -> {
+                    if (mPlaylistService != null) {
+                        if (moveOrCopyModel.getPlaylistOptions()
+                                        == PlaylistOptions.MOVE_PLAYLIST_ITEM
+                                || moveOrCopyModel.getPlaylistOptions()
+                                        == PlaylistOptions.MOVE_PLAYLIST_ITEMS) {
+                            for (PlaylistItemModel playlistItem : moveOrCopyModel.getItems()) {
+                                mPlaylistService.moveItem(playlistItem.getPlaylistId(),
+                                        moveOrCopyModel.getToPlaylistId(), playlistItem.getId());
+                            }
+                            if (moveOrCopyModel.getItems().size() > 0) {
+                                loadPlaylist(moveOrCopyModel.getItems().get(0).getPlaylistId());
+                            }
+                        } else {
+                            String[] playlistIds = new String[moveOrCopyModel.getItems().size()];
+                            for (int i = 0; i < moveOrCopyModel.getItems().size(); i++) {
+                                playlistIds[i] = moveOrCopyModel.getItems().get(i).getId();
+                            }
+                            mPlaylistService.copyItemToPlaylist(
+                                    playlistIds, moveOrCopyModel.getToPlaylistId());
+                        }
+                    }
+                });
+
+        mPlaylistViewModel.getPlaylistOption().observe(
+                PlaylistHostActivity.this, playlistOptionsModel -> {
+                    if (mPlaylistService != null) {
+                        PlaylistOptions option = playlistOptionsModel.getOptionType();
+                        if (option == PlaylistOptions.REMOVE_PLAYLIST_OFFLINE_DATA) {
+                            if (mPlaylistService != null
```

Comment:
at this line `if (mPlaylistService != null` is never true, because line 195 already checked it

Reply:
The same at https://github.com/brave/brave-core/pull/16793/files#diff-d06bc175c295669e39d61fbad1482afad09d7a37ceb55669b1937e2d485c49bbR204 and 
https://github.com/brave/brave-core/pull/16793/files#diff-d06bc175c295669e39d61fbad1482afad09d7a37ceb55669b1937e2d485c49bbR217

Reply:
the same for L236 and L246